datatypes/src/raster/operations/map_elements.rs

```diff
@@ -0,0 +1,364 @@
+use crate::raster::{
+    data_type::DefaultNoDataValue, EmptyGrid, Grid, GridOrEmpty, GridSize, NoDataValue,
+    RasterTile2D,
+};
+use rayon::iter::{IndexedParallelIterator, IntoParallelRefIterator, ParallelIterator};
+
+pub trait MapElements<In, Out, F: Fn(In) -> Option<Out>> {
+    type Output;
+    fn map_elements(self, map_fn: F, out_no_data: Option<Out>) -> Self::Output;
```

Comment:
does it make sense to call the function for every pixel or would it make more sense to have a function that maps all pixels at once?

Reply:
the idea was that this trait takes a fn and applies it to all pixels. so you only have to provide a fn for each pixel.